protocol/network/messages.md

```diff
@@ -1,11 +1,12 @@
 # P2P Network Message
 
-The Bitcoin Cash Peer-to-Peer (P2P) Network protocol is a binary protocol used by Full Nodes and [SPV](/protocol/simple-payment-verification) Nodes, transmitted via TCP.
-The P2P network is similar to a gossip network, where nodes listen for messages and then relays that message to its other peers if it believes that message's content is valid.
+The Bitcoin Cash Peer-to-Peer (P2P) Network protocol is a binary protocol used by Full Nodes and [SPV](/protocol/simple-payment-verification) Nodes, transmitted over TCP/IP.
+Individual nodes on the Bitcoin Cash network connect and create a mesh network where each node is indirectly connected to many others via just a couple of hops.
+In the original Satoshi implementation of the P2P protocol the design of INV and getdata have been used for propagating transaction data using the rules of the gossip protocol values: forwarding validated transactions to a few peer-nodes who send it to others until the entire network has the transaction. This emergent behavior of the P2P layer allows fast propagation without undue strain on any individual node.
 
-P2P network messages do not necessarily have a reply and there is no way to unambiguously connect a sent message to a reply, although many communications are often request/response pairs.
-Nodes may handle incoming messages in parallel, so a message reply order cannot be assumed.
-Messages that cannot be fulfilled are sometimes dropped with no reply, and sometimes replied to via a `reject` message.  
+The P2P protocol is designed around messages. Each message is separate and self-contained. Nodes should be tolerant of message-types they do not understand. It is best to simply ignore those.
+Detailed descriptions of the messages follows below. Generally speaking, each message is an event that the node can choose to respond to. Events range from notifications of new data (transactions/blocks/etc) and
+actual requests for such data to be send and last the actual data being sent. Or, in some specific cases a `reject` message.
```

Comment:
edit for clarity.  Also I think that its important to indicate that implementation may or may not 'reject'.

"Events can be notifications of new data (transactions/blocks/etc), requests for such data to be sent, and the actual data. Or, in some specific cases and implementations a `reject` message.  Since the `reject` message is optional, clients must not rely upon it, instead using it solely for early activation of no-reply timeout logic."